esmvalcore/preprocessor/_area.py

```diff
@@ -440,108 +456,156 @@ def _correct_coords_from_shapefile(cube, cmor_coords, pad_north_pole,
+
+    return requested_geometries
 
-def _geometry_matches_ids(geometry: dict, ids: list):
-    """Returns True if `geometry` matches one of the `ids`."""
-    props = geometry['properties']
 
-    geom_id = [props.get(key, None) for key in SHAPE_ID_KEYS]
-    geom_id = [key for key in geom_id if key is not None]
+def _get_masks_from_geometries(
+    geometries: dict[str, dict],
+    lon: np.ndarray,
+    lat: np.ndarray,
+    method: str = 'contains',
+    decomposed: bool = False,
+) -> dict[str, np.ndarray]:
+    """Get cube masks from requested regions."""
+    if method not in {'contains', 'representative'}:
+        raise ValueError(
+            "Invalid value for `method`. Choose from 'contains', ",
+            "'representative'.")
 
-    if not geom_id:
-        raise KeyError(f'{props} dict has no `name` or `id` key')
+    masks = {}
+    for (id_, geometry) in geometries.items():
+        masks[id_] = _get_single_mask(lon, lat, method, geometry)
 
-    geom_id = geom_id[0]
+    if not decomposed and len(masks) > 1:
+        return _merge_masks(masks, lat.shape)
 
-    return geom_id in ids
+    return masks
 
 
-def _get_bounds(geometries, ids=None):
-    """Get bounds from the subset of geometries defined by `ids`.
+def _get_bounds(
+    geometries: dict[str, dict],
+) -> tuple[float, float, float, float]:
+    """Get bounds from given geometries.
 
     Parameters
     ----------
-    geometries : fiona.Collection
+    geometries: fiona.collection.Collection
         Fiona collection of shapes (geometries).
-    ids : tuple of str, optional
-        List of ids to select from geometry collection. If None,
-        return global bounds (``geometries.bounds``)
 
     Returns
     -------
     lat_min, lon_min, lat_max, lon_max
-        Returns coordinates deliminating bounding box for shape ids.
-    """
-    if not ids:
-        return geometries.bounds
-
-    subset = [geom for geom in geometries if _geometry_matches_ids(geom, ids)]
+        Coordinates deliminating bounding box for shape ids.
 
-    all_bounds = np.vstack([fiona.bounds(geom) for geom in subset])
+    """
+    all_bounds = np.vstack(
+        [fiona.bounds(geom) for geom in geometries.values()]
+    )
     lon_max, lat_max = all_bounds[:, 2:].max(axis=0)
     lon_min, lat_min = all_bounds[:, :2].min(axis=0)
 
     return lon_min, lat_min, lon_max, lat_max
 
 
-def _get_shape(lon, lat, method, item):
-    shape = shapely.geometry.shape(item['geometry'])
+def _get_single_mask(
+    lon: np.ndarray,
+    lat: np.ndarray,
+    method: str,
+    geometry: dict,
+) -> np.ndarray:
+    """Get single mask from one region."""
+    shape = shapely.geometry.shape(geometry['geometry'])
     if method == 'contains':
-        select = shapely.vectorized.contains(shape, lon, lat)
-    if method == 'representative' or not select.any():
-        select = _select_representative_point(shape, lon, lat)
-    return select
+        mask = shapely.vectorized.contains(shape, lon, lat)
```

Comment:
this works in the new shapely >2 too right?

Reply:
No idea, I guess we will see once the tests are run? Do we get shapely > 2 in our CI env now?

Reply:
I think so on Circle yes, which makes my question irrelevant :grin:

Reply:
Tests ran fine again, so I guess we are safe here 🚀